File: Container Server/backend/models/post.js
/**************************************************************************************************
 The model of the Posts used as a model of the packing list received and sent to the user and as a 
 model of the data result from the algoritm that will be returned to the front end.
 **************************************************************************************************/
const mongoose = require('mongoose');


const postSchema = mongoose.Schema({
label: {type: String, required: true},
packingList:{type:mongoose.Schema.Types.Array},
container: {type: String, required: true},
creator: {type: mongoose.Schema.Types.ObjectId, ref: "User", required: true} //to reffer the post to the specific user who created it
});


module.exports = mongoose.model("Post", postSchema);

File: Container/Container/Models/Post.swift
//
//  Post.swift
//  Container
//
//  Created by MacBook on 2021/3/14.
//
//*********************************************************************************
//**            model of a post of a packing list data for communicating with    **
//**            the server                                                       **
//*********************************************************************************
import Foundation
struct Post: Codable{
    var label: String
    var packingList: [PackageData]
    var container: String
    
    init(_ label: String, _ packingList: [PackageData], _ container: String){
        self.label = label
        self.packingList = packingList
        self.container = container
    }
}
